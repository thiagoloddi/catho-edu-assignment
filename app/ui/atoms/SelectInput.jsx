import React from 'react';
import injectSheet from 'react-jss';
import propTypes from 'prop-types';

const style = {
  select: {
    padding: '5px',
    backgroundColor: 'white',
    outline: 'none'
  }
}

const SelectInput = ({ classes, options, className = '', value, onChange }) => {

  return (
    <select onChange={onChange} value={value} className={`${classes.select} ${className}`}>
      {options.map(({ id, label }) => 
        <option key={id} value={id}>{label}</option>
      )}
    </select>
  )
}

SelectInput.propTypes = {
	options: propTypes.arrayOf(propTypes.shape({
		id: propTypes.string.isRequired,
		label: propTypes.string.isRequired
	})).isRequired,
	value: propTypes.string.isRequired,
	onChange: propTypes.func.isRequired,
	className: propTypes.string
};

SelectInput.defaultProps = {
	options: [],
	className: ''
};

export default injectSheet(style)(SelectInput);