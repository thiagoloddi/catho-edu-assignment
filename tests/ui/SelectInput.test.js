import React from 'react';
import renderer from 'react-test-renderer';
import SelectInput from '../../app/ui/atoms/SelectInput';

const func = () => {};

it('renders correctly', () => {
  const tree = renderer
    .create(<SelectInput value={'1'} onChange={func} options={[
			{ id: '1', label: 'opt 1' },
			{ id: '2', label: 'opt 2' }
		]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});