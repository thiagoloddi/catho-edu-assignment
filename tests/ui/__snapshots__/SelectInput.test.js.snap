// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<select
  className="SelectInput-select-0-1-1 "
  onChange={[Function]}
  value="1"
>
  <option
    value="1"
  >
    opt 1
  </option>
  <option
    value="2"
  >
    opt 2
  </option>
</select>
`;
